src/Common/tests/Collections/IEnumerableTest.cs

```diff
@@ -0,0 +1,428 @@
+                                   enumerator.Reset();
+                               }
+                               else
+                               {
+                                   VerifyEnumerator(items, enumerator);
+                               }
+                           }, 5);
+            }
+        }
+
+        [Fact]
+        public void ModifyCollectionWithNewEnumerator()
+        {
+            var enumerable = GetEnumerable(GenerateItems(16));
+            var enumerator = enumerable.GetEnumerator();
+            InvalidateEnumerator(enumerable);
+            VerifyModifiedEnumerator(enumerator, null, true, false);
+        }
+
+        [Fact]
+        public void EnumerateFirstItemThenModify()
+        {
+            var items = GenerateItems(16);
+            var enumerable = GetEnumerable(items);
+            var enumerator = enumerable.GetEnumerator();
+            VerifyEnumerator(items, enumerator, 0, 1, false, true);
+            var currentItem = enumerator.Current;
+            InvalidateEnumerator(enumerable);
+            VerifyModifiedEnumerator(
+                enumerator,
+                currentItem,
+                false,
+                false);
+        }
+
+        [Fact]
+        public void EnumeratePartOfCollectionThenModify()
+        {
+            var items = GenerateItems(16);
+            var enumerable = GetEnumerable(items);
+            var enumerator = enumerable.GetEnumerator();
+            VerifyEnumerator(items, enumerator, 0, items.Length / 2, false, true);
+
+            var currentItem = enumerator.Current;
+            InvalidateEnumerator(enumerable);
+            VerifyModifiedEnumerator(enumerator, currentItem, false, false);
+        }
+
+        [Fact]
+        public void EnumerateEntireCollectionThenModify()
+        {
+            var items = GenerateItems(16);
+            var enumerable = GetEnumerable(items);
+            var enumerator = enumerable.GetEnumerator();
+            VerifyEnumerator(items, enumerator, 0, items.Length, false, true);
+
+            var currentItem = enumerator.Current;
+            InvalidateEnumerator(enumerable);
+            VerifyModifiedEnumerator(enumerator, currentItem, false, true);
+        }
+
+        [Fact]
+        public void EnumerateThenModifyThrows()
+        {
+            var items = GenerateItems(16);
+            var enumerable = GetEnumerable(items);
+            var enumerator = enumerable.GetEnumerator();
+            VerifyEnumerator(items, enumerator, 0, items.Length / 2, false, true);
+
+            var currentItem = enumerator.Current;
+            InvalidateEnumerator(enumerable);
+            Assert.Equal(currentItem, enumerator.Current);
+            Assert.Throws<InvalidOperationException>(
+                () => enumerator.MoveNext());
+            Assert.Throws<InvalidOperationException>(
+                () => enumerator.Reset());
+        }
+
+        [Fact]
+        [ActiveIssue(1170)]
```

Comment:
I don't believe there is a way to get xunit to do that. It looks like Fact is inherited but I'm not sure how xunit searches for methods. A virtual method with fact on it may result in two test cases.